### DISCUSSION TITLE
Reproducing R built graph in cytoscape.js

### DISCUSSION THREAD

**Discussion by zSchadt** [cytoscape/cytoscape.js#3003] [Q&A]
I came across this cool example: https://github.com/idekerlab/cy-rest-R ([specifically this script](https://github.com/idekerlab/cy-rest-R/blob/develop/workflow1_structure_based_visualization.R))
I'm attempting to reproduce it in a web project using cytoscape.js but am running into a few issues.

It looks like the above example uses a layout called: "kamada-kawai"
* I can't find a similar layout within cytoscape.js, maybe it has a different name?
* are other generated statistics like closeness used within the layout function somewhere? I haven't been able to find any mention of "closeness" in the cytoscape.js documentation, but in the above example closeness is assigned to every node and then never "read" again.

**What I've tried:**
* layout type: 'cose'
* separately used igraph within python to generate the community data

--- 

Should I be able to fully reproduce this example (or any cytoscape R output for that matter) using cytoscape.js?

Sorry to just drop a github repo asking for help, but I feel the above example's styling and setup is straight forward; and I'm getting to the point where I'm thinking it's not possible to reproduce using only cystoscape.js.

Thanks in advance for any help!

Edit:

I figured it out. Mostly came down to a lack of graph knowledge. I used ended up using a separate library to generate the kamada-kawai coordinates. Then tweaked the style to my liking.

**Comment by maxkfranz** [comment]
@keiono, any thoughts on reproducing your script?

**Reply by zSchadt** [reply]
Hey, thanks for getting back! I ended up figuring it out!
Apologies - should have updated the post sooner.

**Reply by maxkfranz** [reply]
Could you share your results?  The next person with a similar question will appreciate it
